apps/explorer/src/utils/getAmount.ts

```diff
@@ -72,27 +74,26 @@ export function getTransfersAmount(
             : null;
     }
 
-    const paySuiData =
-        getPaySuiTransaction(txnData) ?? getPayTransaction(txnData);
+    const payData = getPaySuiTransaction(txnData) ?? getPayTransaction(txnData);
 
-    const amountByRecipient = paySuiData?.recipients.reduce(
+    const amountByRecipient = payData?.recipients.reduce(
         (acc, value, index) => {
             return {
                 ...acc,
                 [value]: {
                     amount:
-                        paySuiData.amounts[index] +
+                        payData.amounts[index] +
                         (value in acc ? acc[value].amount : 0),
-                    coinType: txnEffect
-                        ? getCoinType(
-                              txnEffect,
-                              paySuiData.recipients[index] ||
-                                  paySuiData.recipients[0]
-                          )
-                        : null,
-                    address:
-                        paySuiData.recipients[index] ||
-                        paySuiData.recipients[0],
+                    // for PaySuiTransaction the coinType is SUI
+                    coinType:
+                        txKindName === 'PaySui'
+                            ? SUI_TYPE_ARG
+                            : getCoinType(
+                                  txnEffect || null,
+                                  payData.recipients[index] ||
```

Comment:
why not just `value` (or `recipientAddress` if you rename it)

Reply:
and I think `payData.recipients[0]` is unnecessary

Reply:
done